src/shogun/base/SGObject.h

```diff
@@ -533,10 +534,10 @@ class CSGObject
 	 */
 	virtual std::string to_string() const;
 
-	/** Returns set of all parameter names of the object.
+	/** Returns map of all parameter names and AnyParameter references of the object.
 	 *
 	 */
-	std::vector<std::string> parameter_names() const;
+	std::map<std::string, std::shared_ptr<const AnyParameter>> get_parameters() const;
```

Comment:
use Some instead of shared_ptr... that you can use in any swig language :)

Reply:
I don't think I can because AnyParameter isn't derived from CSGObject, so would have to implement it's own ref counter..

Reply:
sooo true 🤦‍♂️

Reply:
But this isn't meant to be exposed directly to SWIG anyway, so it should be fine no?

Reply:
The idea is to then use this to extend SGObject in shogun.i to get some information about the parameters, i.e. name and type

Reply:
ok so then it's a go... but in that case u could use std::map as well ;)

Reply:
The CMap is still needed for the SWIG interface in `CMap<std::string, std::string> parameter_types() const`, right?

Reply:
what's the value content?

Reply:
the std::string is the demangled type for the param (key), which is then used for now in an error message in the Python API

Reply:
hide it from swig as well `#ifndef SWIG // SWIG should skip this part`